### ISSUE TITLE
Migrate CI to GitHub Actions?

### ISSUE THREAD

**Issue by jacobtomlinson** [dask/dask-xgboost#79]
Due to changes in the Travis CI billing, the Dask org is migrating Travis CI to GitHub Actions.

This repo appears to use CircleCI. As we are putting in the effort to migrate many projects to GitHub Actions does it make sense to standardise here?

See dask/community#107 for more details.

**Comment by jameslamb** [comment]
Instead of anyone spending time on this, I think it's worth waiting a little longer for `xgboost.dask` to be fully ready to replace `dask-xgboost` (https://github.com/dask/community/issues/104)

XGBoost maintainers have a pull request up right now for the last remaining issue from https://github.com/dask/dask-xgboost/issues/39 that is needed for `xgboost.dask` to be usable as a full replacement for `dask-xgboost` (https://github.com/dmlc/xgboost/pull/6343)

I'd hate to see someone spend a lot of time moving this project's CI just for it to be abandoned in a few months.

If you disagree and there's something I haven't thought about, I'd be happy to do the work to move it to GitHub Actions.

**Comment by jakirkham** [comment]
My guess is Jacob was opening issues across all of the Dask repos. So probably wasn't thinking of the specifics of each repo. Feel free to correct me if that is incorrect Jacob 🙂

That said, yeah agree that we don't need to worry about that change on this repo for now. If we need to update this repo again in the future, we might want to revisit at that time

**Comment by jacobtomlinson** [comment]
Yeah I was just opening issues on all affected repos in order to start discussions and track things.

I propose we leave this open pending the upstream work. I assume once `xgboost.dask` is ready this repo will be archived anyway.
